### ISSUE TITLE
DetailsList GroupHeader checkbox is missing aria-label 

### ISSUE THREAD

**Issue by karanbirsingh** [OfficeDev/office-ui-fabric-react#7751]
This is the same issue as #5986 which was closed with a different solution. I think we may still want the aria-label.

### Environment Information 
 
- __Package version(s)__: 6.129.1
- __Browser and OS versions__: Chrome (Windows 10) Version 71.0.3578.98 (Official Build) (64-bit)

### Please provide a reproduction of the bug in a codepen:

Reusing this [codepen ](
https://codepen.io/peterliumich/pen/jvNWRy?editors=1111)from #5986 
 
#### Actual behavior:
 
The checkbox-button does not have an aria-label.
 
#### Expected behavior:

The checkbox-button has an aria-label. 

In the linked PR [discussion](https://github.com/OfficeDev/office-ui-fabric-react/pull/5987) the following examples are linked which don't use aria-labels in this case:

> Here are the examples I was looking at:
> - https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Techniques/Using_the_checkbox_role
> - https://www.w3.org/TR/2016/WD-wai-aria-practices-1.1-20160317/examples/checkbox/checkbox-2.html

Guru (@iamrafan) mentioned that the linked examples are a bit misleading because the checkbox has existing content a screen reader can use. In our OfficeFabric case, the checkbox has no content so an aria-label would provide some helpful context.

Maybe something like `expandButtonProps` here would be helpful for the checkbox:
https://github.com/OfficeDev/office-ui-fabric-react/blob/59276033edb69b6e845dde243fa3e9fce2bc0f19/packages/office-ui-fabric-react/src/components/GroupedList/GroupHeader.base.tsx#L21

#### Priorities and help requested:
 
Are you willing to submit a PR to fix? (Yes, No) see PR [here](https://github.com/OfficeDev/office-ui-fabric-react/pull/7777)
Requested priority: (Blocking, High, Normal, Low)  normal
